### ISSUE TITLE
Agent Tools dialog: Close button missing accessible name

### ISSUE THREAD

**Issue by kangabell** [danny-avila/LibreChat#5394]
There is no screen reader accessible label on this button. Additionally, the color contrast ratio when in dark mode is too low (#595959 on #595959)

## WCAG Criteria
[2.5.3 Label in Name - A](https://www.w3.org/WAI/WCAG21/Understanding/label-in-name)

## Action
Add visible text to the button, or an aria-label. Also lighten the color in dark mode to at least #6B6B6B.

## Notes
The missing label is an issue identified by Accessibility issues for screenreader users #5187
The contrast in light mode looks fine.

## Screenshot

![Dialog titled "Agent Tools" with a letter "x" icon in the far right.](https://github.com/user-attachments/assets/defe4687-0668-42b2-82f2-8be7ef88e0e7)

## Relevant Code
```html
<button class="inline-block text-text-tertiary hover:text-text-secondary" tabindex="0">
  <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="lucide lucide-x">
    <path d="M18 6 6 18"></path>
    <path d="m6 6 12 12"></path>
  </svg>
</button>
